wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java

```diff
@@ -37,4 +41,36 @@
     String expected = "/* comment inside comment. &#42;/";
     assertThat(JavaGenerator.sanitizeJavadoc(input)).isEqualTo(expected);
   }
+
+  @Test public void generateProtoFields() {
+    Schema schema = new SchemaBuilder()
+        .add("message.proto", ""
+            + "import \"foo.proto\";\n"
+            + "message Message {\n"
+            + "  optional foo_package.Foo field = 1;\n"
+            + "  map<string, foo_package.Bar> bars = 2;\n"
+            + "  repeated int32 numbers = 3;\n"
+            + "}\n")
+        .add("foo.proto", ""
+            + "package foo_package;\n"
+            + "message Foo {\n"
+            + "}\n"
+            + "message Bar {\n"
+            + "}\n")
+        .build();
+
+    MessageType message = (MessageType) schema.getType("Message");
+
+    JavaGenerator javaGenerator = JavaGenerator.get(schema);
+    TypeSpec typeSpec = javaGenerator.generateProtoFields(message);
+
+    assertThat(typeSpec.toString()).isEqualTo(""
```

Comment:
There’s a slightly better way to do the `toString()` thing that gets you imports. It’ll make this way more readable.

You probably want this method:
https://github.com/square/javapoet/blob/master/src/test/java/com/squareup/javapoet/TypeSpecTest.java#L1511